### PR TITLE
Add initial /user api endpoint

### PR DESCRIPTION
**Pull request by kevinburleigh75**
@philschatz @jpslav @Dantemss

### CODE REVIEW

**Comment by philschatz** [comment]
+1 Does it yield JSON containing the current user's id?

File: spec/controllers/api/v1/users_controller_spec.rb [hunk 1]
```diff
@@ -4,10 +4,25 @@
...
-                                              application: application, 
+  let!(:user_1_token)    { FactoryGirl.create :doorkeeper_access_token,
+                                              application: application,
                                               resource_owner_id: user_1.id }
 
+  describe "user" do
+    context "caller has an authorization token" do
+      it "should return a success (200) code" do
+        api_get :user, user_1_token
+        expect(response.code).to eq('500')
```
Comment (Dantemss) [inline spec/controllers/api/v1/users_controller_spec.rb]: Forgot this one here :)

**Comment by Dantemss** [comment]
@philschatz Right now it only seems to return 200 or 403.

**Comment by jpslav** [comment]
Phil you told me just 200 or 403 was fine for now so that's what I told KB

File: spec/controllers/api/v1/users_controller_spec.rb [hunk 2]
```diff
@@ -1,14 +1,34 @@
 require "rails_helper"
 
-describe Api::V1::UsersController, :type => :controller, :api => true, :version => :v1 do
+describe Api::V1::UsersController, :api => true, :version => :v1 do
```
Comment (Dantemss) [inline spec/controllers/api/v1/users_controller_spec.rb]: Could leave the `:type => :controller` in here (so this spec runs when you call `rake spec:controllers`)

File: spec/controllers/api/v1/users_controller_spec.rb [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
   let!(:application)     { FactoryGirl.create :doorkeeper_application }
   let!(:user_1)          { FactoryGirl.create :user }
-  let!(:user_1_token)    { FactoryGirl.create :doorkeeper_access_token, 
-                                              application: application, 
+  let!(:user_1_token)    { FactoryGirl.create :doorkeeper_access_token,
+                                              application: application,
                                               resource_owner_id: user_1.id }
 
-  describe "tasks" do
+  describe "#show", type: :request do
```
Comment (Dantemss) [inline spec/controllers/api/v1/users_controller_spec.rb]: Would be interesting to see a test case for a non-user (client credentials) token.

**Comment by Dantemss** [comment]
Looks good.

**Pull request merged**